zerver/lib/send_email.py

```diff
@@ -147,23 +137,25 @@ def send_future_email(template_prefix: str, realm: Realm, to_user_ids: Optional[
         # For logging the email
 
     assert (to_user_ids is None) ^ (to_emails is None)
-    if to_user_ids is not None:
-        # The realm is redundant if we have a to_user_id; this assert just
-        # expresses that fact
-        assert(len(to_user_ids) == 1)
-        assert(UserProfile.objects.filter(id__in=to_user_ids, realm=realm).exists())
-        to_field = {'user_id': to_user_ids[0]}  # type: Dict[str, Any]
-    else:
-        assert to_emails is not None
-        assert(len(to_emails) == 1)
-        to_field = {'address': parseaddr(to_emails[0])[1]}
-
-    ScheduledEmail.objects.create(
+    email = ScheduledEmail.objects.create(
         type=EMAIL_TYPES[template_name],
         scheduled_timestamp=timezone_now() + delay,
         realm=realm,
-        data=ujson.dumps(email_fields),
-        **to_field)
+        data=ujson.dumps(email_fields))
+    try:
+        if to_user_ids is not None:
+            users = UserProfile.objects.filter(id__in=to_user_ids, realm=realm)
+            assert(users.exists())
+            for user in users:
+                email.users.add(user)
```

Comment:
I think there's probably a Django way to just add these by ID, rather than fetching the full UserProfile objects.